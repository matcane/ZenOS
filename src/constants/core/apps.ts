import calc from "@/config/apps/calc.json";
import camera from "@/config/apps/camera.json";
import chrome from "@/config/apps/chrome.json";
import clock from "@/config/apps/clock.json";
import messages from "@/config/apps/messages.json";
import phone from "@/config/apps/phone.json";
import settings from "@/config/apps/settings.json";

type TNavigationBar = {
  backgroundColor: string;
  dark: { backgroundColor: string };
};

export type TApp = {
  name: string;
  slug: string;
  version: string;
  icon: string;
  navigationBar?: TNavigationBar;
};

export const APPS = [settings, calc, clock];

export const PINNED_APPS = [phone, messages, chrome, camera];

export const APPS_ICON: Record<string, any> = {
  "@assets/icons/icons8-settings.webp": require("@assets/icons/icons8-settings.webp"),
  "@assets/icons/icons8-calculator.webp": require("@assets/icons/icons8-calculator.webp"),
  "@assets/icons/icons8-clock.webp": require("@assets/icons/icons8-clock.webp"),
  "@assets/icons/icons8-phone.webp": require("@assets/icons/icons8-phone.webp"),
  "@assets/icons/icons8-messages.webp": require("@assets/icons/icons8-messages.webp"),
  "@assets/icons/icons8-chrome.webp": require("@assets/icons/icons8-chrome.webp"),
  "@assets/icons/icons8-camera.webp": require("@assets/icons/icons8-camera.webp"),
};

export const APPS_ICON_SOURCE: Record<string, any> = {
  "@assets/icons/icons8-settings.webp": "https://icons8.com/icon/4UC7V4crVOgk/settings",
  "@assets/icons/icons8-calculator.webp": "https://icons8.com/icon/w8uA2lyOFNbd/calculator",
  "@assets/icons/icons8-clock.webp": "https://icons8.com/icon/kZYJCScZSxUH/clock",
  "@assets/icons/icons8-phone.webp": "https://icons8.com/icon/8U0YyPNowmTt/phone",
  "@assets/icons/icons8-messages.webp": "https://icons8.com/icon/U2maxZvgoDog/messages",
  "@assets/icons/icons8-chrome.webp": "https://icons8.com/icon/QBljOALaVp01/chrome",
  "@assets/icons/icons8-camera.webp": "https://icons8.com/icon/eKEI8q7Hjioq/camera",
};
