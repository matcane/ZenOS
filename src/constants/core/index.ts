export * from "./apps";
export * from "./settingsFields";
