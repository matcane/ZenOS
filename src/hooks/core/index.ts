export * from "./useTheme";
export * from "./useTime";
export * from "./useTimer";
