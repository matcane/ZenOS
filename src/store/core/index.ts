export * from "./themeStore";
export * from "./authStore";
export * from "./wallpaperStore";
