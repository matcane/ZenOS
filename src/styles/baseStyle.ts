import { StyleSheet } from "react-native";

export const baseStyle = StyleSheet.create({
  transparent: { backgroundColor: undefined },

  fontXS: { fontSize: 12 },
  fontSM: { fontSize: 16 },
  fontSMD: { fontSize: 24 },
  fontMD: { fontSize: 32 },
  fontLG: { fontSize: 48 },
  fontXL: { fontSize: 64 },
  fontXXL: { fontSize: 128 },

  heightSM: { height: 16 },
  heightMD: { height: 32 },
  heightLG: { height: 48 },
  heightXL: { height: 64 },
  heightXXL: { height: 128 },

  widthSM: { width: 16 },
  widthMD: { width: 32 },
  widthLG: { width: 48 },
  widthXL: { width: 64 },
  widthXXL: { width: 128 },

  sizeSM: { width: 16, height: 16 },
  sizeMD: { width: 32, height: 32 },
  sizeLG: { width: 48, height: 48 },
  sizeXL: { width: 64, height: 64 },

  static: { position: "static" },
  absolute: { position: "absolute" },
  relative: { position: "relative" },

  rounded: { borderRadius: 5 },
  roundedMD: { borderRadius: 10 },
  roundedLG: { borderRadius: 20 },
  roundedFull: { borderRadius: 999 },

  flex: { flex: 1 },
  flexNone: { flex: 0 },
  flexGrow: { flexGrow: 1 },
  flexGrowNone: { flexGrow: 0 },
  flexShrink: { flexShrink: 1 },
  flexShrinkNone: { flexShrink: 0 },

  flexRow: { flexDirection: "row" },
  flexRowReverse: { flexDirection: "row-reverse" },
  flexCol: { flexDirection: "column" },
  flexColReverse: { flexDirection: "column-reverse" },

  paddingSM: { padding: 5 },
  paddingMD: { padding: 10 },
  paddingLG: { padding: 15 },
  paddingXL: { padding: 20 },

  paddingHorizontalSM: { paddingHorizontal: 10 },
  paddingHorizontalMD: { paddingHorizontal: 20 },
  paddingHorizontalLG: { paddingHorizontal: 30 },
  paddingHorizontalXL: { paddingHorizontal: 40 },
  paddingHorizontalXXL: { paddingHorizontal: 90 },

  paddingVerticalSM: { paddingVertical: 10 },
  paddingVerticalMD: { paddingVertical: 20 },
  paddingVerticalLG: { paddingVertical: 30 },
  paddingVerticalXL: { paddingVertical: 40 },
  paddingVerticalXXL: { paddingVertical: 80 },

  paddingTopSM: { paddingTop: 5 },
  paddingTopMD: { paddingTop: 10 },
  paddingTopLG: { paddingTop: 15 },
  paddingTopXL: { paddingTop: 20 },
  paddingTopXXL: { paddingTop: 40 },

  paddingBottomSM: { paddingBottom: 5 },
  paddingBottomMD: { paddingBottom: 10 },
  paddingBottomLG: { paddingBottom: 15 },
  paddingBottomXL: { paddingBottom: 20 },
  paddingBottomXXL: { paddingBottom: 40 },

  paddingLeftSM: { paddingLeft: 5 },
  paddingLeftMD: { paddingLeft: 10 },
  paddingLeftLG: { paddingLeft: 15 },
  paddingLeftXL: { paddingLeft: 20 },
  paddingLeftXXL: { paddingLeft: 40 },

  paddingRightSM: { paddingRight: 5 },
  paddingRightMD: { paddingRight: 10 },
  paddingRightLG: { paddingRight: 15 },
  paddingRightXL: { paddingRight: 20 },
  paddingRightXXL: { paddingRight: 40 },

  marginSM: { margin: 5 },
  marginMD: { margin: 10 },
  marginLG: { margin: 15 },
  marginXL: { margin: 20 },

  marginHorizontalSM: { marginHorizontal: 10 },
  marginHorizontalMD: { marginHorizontal: 20 },
  marginHorizontalLG: { marginHorizontal: 30 },
  marginHorizontalXL: { marginHorizontal: 40 },
  marginHorizontalXXL: { marginHorizontal: 80 },

  marginVerticalSM: { marginVertical: 10 },
  marginVerticalMD: { marginVertical: 20 },
  marginVerticalLG: { marginVertical: 30 },
  marginVerticalXL: { marginVertical: 40 },
  marginVerticalXXL: { marginVertical: 80 },

  marginTopSM: { marginTop: 5 },
  marginTopMD: { marginTop: 10 },
  marginTopLG: { marginTop: 15 },
  marginTopXL: { marginTop: 20 },
  marginTopXXL: { marginTop: 40 },

  marginBottomSM: { marginBottom: 5 },
  marginBottomMD: { marginBottom: 10 },
  marginBottomLG: { marginBottom: 15 },
  marginBottomXL: { marginBottom: 20 },
  marginBottomXXL: { marginBottom: 40 },

  marginLeftSM: { marginLeft: 5 },
  marginLeftMD: { marginLeft: 10 },
  marginLeftLG: { marginLeft: 15 },
  marginLeftXL: { marginLeft: 20 },
  marginLeftXXL: { marginLeft: 40 },

  marginRightSM: { marginRight: 5 },
  marginRightMD: { marginRight: 10 },
  marginRightLG: { marginRight: 15 },
  marginRightXL: { marginRight: 20 },
  marginRightXXL: { marginRight: 40 },

  justifyStart: { justifyContent: "flex-start" },
  justifyEnd: { justifyContent: "flex-end" },
  justifyCenter: { justifyContent: "center" },
  justifyBetween: { justifyContent: "space-between" },
  justifyAround: { justifyContent: "space-around" },
  justifyEvenly: { justifyContent: "space-evenly" },

  contentStart: { alignContent: "flex-start" },
  contentEnd: { alignContent: "flex-end" },
  contentCenter: { alignContent: "center" },
  contentBetween: { alignContent: "space-between" },
  contentAround: { alignContent: "space-around" },
  contentEvenly: { alignContent: "space-evenly" },

  itemsStart: { alignItems: "flex-start" },
  itemsEnd: { alignItems: "flex-end" },
  itemsCenter: { alignItems: "center" },
  itemsBaseline: { alignItems: "baseline" },
  itemsStretch: { alignItems: "stretch" },

  textLeft: { textAlign: "left" },
  textCenter: { textAlign: "center" },
  textRight: { textAlign: "right" },
  textJustify: { textAlign: "justify" },
});
