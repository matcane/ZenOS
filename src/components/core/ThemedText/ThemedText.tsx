import { Text, TextProps } from "react-native";

import { useThemeColor } from "@/hooks/useTheme";

export default function ThemedText({ style, ...rest }: TextProps) {
  const theme = useThemeColor();

  return <Text style={[{ color: theme.text }, style]} {...rest} />;
}
