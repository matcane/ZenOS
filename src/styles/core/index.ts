import { StyleSheet } from "react-native";

import { baseStyle } from "../baseStyle";

export const coreStyles = StyleSheet.create({
  calcDisplay: {
    ...baseStyle.itemsEnd,
    ...baseStyle.justifyEnd,
    ...baseStyle.paddingBottomXL,
  },

  calcDisplayText: { ...baseStyle.paddingHorizontalMD },

  stopwatchStartStop: { right: "50%", marginRight: -30, bottom: 0, marginBottom: 30 },

  stopwatchReset: { right: "25%", marginRight: -20, bottom: 0, marginBottom: 35 },

  actionButton: {
    ...baseStyle.absolute,
    ...baseStyle.roundedFull,
    ...baseStyle.justifyCenter,
    ...baseStyle.itemsCenter,
    elevation: 5,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },

  container: {
    ...baseStyle.paddingXL,
    ...baseStyle.itemsCenter,
  },

  navigationBarContainer: {
    ...baseStyle.justifyEvenly,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexRow,
    height: 45,
    gap: 20,
  },

  backNavButton: { transform: [{ rotate: "30deg" }] },

  settingsFieldItem: {
    ...baseStyle.paddingLeftMD,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexGrow,
    height: 60,
    gap: 10,
  },

  firstSettingsFieldItem: {
    borderTopEndRadius: 10,
    borderTopStartRadius: 10,
  },

  lastSettingsFieldItem: {
    borderBottomEndRadius: 10,
    borderBottomStartRadius: 10,
  },

  settingsFieldIcon: {
    ...baseStyle.sizeLG,
  },

  statusBarContainer: {
    ...baseStyle.absolute,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.paddingHorizontalMD,
    height: 32,
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 999,
  },

  statusBarIcons: {
    ...baseStyle.flexRow,
    gap: 2,
  },

  wallpaper: {
    ...baseStyle.flexGrow,
    ...baseStyle.justifyCenter,
    ...baseStyle.paddingTopXL,
  },
});
