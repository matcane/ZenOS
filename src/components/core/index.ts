import ActionButton from "./ActionButton/ActionButton";
import AppWidget from "./AppWidget/AppWidget";
import AuthForm from "./AuthForm/AuthForm";
import CalcButton from "./CalcButton/CalcButton";
import NavigationBar from "./NavigationBar/NavigationBar";
import SafeArea from "./SafeArea/SafeArea";
import SettingsField from "./SettingsField/SettingsField";
import StatusBar from "./StatusBar/StatusBar";
import ThemedText from "./ThemedText/ThemedText";
import ThemedView from "./ThemedView/ThemedView";
import Wallpaper from "./Wallpaper/Wallpaper";

export {
  NavigationBar,
  StatusBar,
  ThemedText,
  ThemedView,
  Wallpaper,
  ActionButton,
  AppWidget,
  CalcButton,
  SettingsField,
  SafeArea,
  AuthForm,
};
