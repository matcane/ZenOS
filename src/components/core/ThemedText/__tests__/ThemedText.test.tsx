import { render } from "@testing-library/react-native";

import { useTheme } from "@/hooks/core";
import { baseStyle } from "@/styles/baseStyle";

import ThemedText from "../ThemedText";

jest.mock("@/hooks/core", () => ({
  useTheme: jest.fn(),
}));

const { fontSM } = baseStyle;

describe("ThemedText component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    (useTheme as jest.Mock).mockReturnValue({ text: "#000" });

    const { toJSON } = render(<ThemedText />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("applies the correct text color from the theme", () => {
    const mockTheme = { text: "#ff0000" };
    (useTheme as jest.Mock).mockReturnValue(mockTheme);

    const { getByTestId } = render(<ThemedText testID="themed-text" />);

    const themedText = getByTestId("themed-text");
    expect(themedText.props.style).toContainEqual({ color: mockTheme.text });
  });

  it("applies additional styles passed via props", () => {
    (useTheme as jest.Mock).mockReturnValue({ text: "#000" });

    const { getByTestId } = render(<ThemedText style={fontSM} testID="themed-text" />);

    const themedText = getByTestId("themed-text");
    expect(themedText.props.style).toContainEqual(fontSM);
  });

  it("renders children correctly", () => {
    (useTheme as jest.Mock).mockReturnValue({ text: "#000" });

    const { getByText } = render(<ThemedText>Child text</ThemedText>);

    expect(getByText("Child text")).toBeTruthy();
  });
});
