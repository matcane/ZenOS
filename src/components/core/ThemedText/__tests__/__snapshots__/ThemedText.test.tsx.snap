// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedText component renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      {
        "fontSize": 16,
      },
      undefined,
    ]
  }
/>
`;
